src/DataGenerator.cpp

```diff
@@ -0,0 +1,499 @@
+             al_it != alternative_node.end(); ++al_it) {
+
+          // std::cout << "crit : " << al_it->child_value() << std::endl;
+          if (strcmp(al_it->child_value(), "assignment") != 0) {
+            float perf = atof(al_it->child_value());
+            alt_perf.push_back(perf);
+          }
+        }
+        break;
+      }
+    }
+    return Performance(alt_id, criteria, alt_perf);
+  }
+}
+
+std::vector<std::string>
+DataGenerator::getAlternativeIds(std::string fileName) {
+  pugi::xml_document doc;
+  std::string path = "../data/" + fileName;
+  if (!doc.load_file(path.c_str()))
+    throw std::invalid_argument("Cannot open xml file, please check path");
+
+  if (DataGenerator::getXmlFileType(fileName) == "model") {
+    // if we have a model xml
+    throw std::invalid_argument("Cannot find any alternatives in xml file, "
+                                "most likely have a xml model file");
+  } else {
+    // creating vector object to store alternative ids
+    std::vector<std::string> alt_ids;
+
+    pugi::xml_node node_dataset = doc.child("dataset");
+    for (pugi::xml_node_iterator it = node_dataset.begin();
+         it != node_dataset.end(); ++it) {
+      if (strcmp(it->name(), "alternative") == 0) {
+        const char *s = it->child_value();
+        std::string str(s);
+        alt_ids.push_back(s);
+      }
+    }
+    return alt_ids;
+  }
+}
+
+std::vector<std::string> DataGenerator::getCriteriaIds(std::string fileName) {
+  pugi::xml_document doc;
+  std::string path = "../data/" + fileName;
+  if (!doc.load_file(path.c_str()))
+    throw std::invalid_argument("Cannot open xml file, please check path");
+
+  // creating vector object to store criteria ids
+  std::vector<std::string> crit_ids;
+
+  if (DataGenerator::getXmlFileType(fileName) == "model") {
+    // if we have a model xml
+    // going from root ie model then to its first child modelName and passing
+    // to the 5th child ie first criterion
+    pugi::xml_node node_model = doc.child("model");
+
+    for (pugi::xml_node_iterator it = node_model.begin();
+         it != node_model.end(); ++it) {
+      if (strcmp(it->name(), "modelName") != 0 &&
+          strcmp(it->name(), "criteria") != 0 &&
+          strcmp(it->name(), "categories") != 0 &&
+          strcmp(it->name(), "lambda") != 0) {
+        const char *s = it->name();
+        std::string str(s);
+        crit_ids.push_back(s);
+      }
+    }
+  } else {
+    // if we have a dataset xml
+    pugi::xml_node node_dataset = doc.child("dataset").child("alternative");
+    for (pugi::xml_node_iterator it = node_dataset.begin();
+         it != node_dataset.end(); ++it) {
+      if (strcmp(it->name(), "") != 0 & strcmp(it->name(), "assignment") != 0) {
+        const char *s = it->name();
+        std::string str(s);
+        crit_ids.push_back(s);
+      }
+    }
+  }
+  return crit_ids;
+}
+
+Criterion DataGenerator::getCriterion(std::string fileName,
```

Comment:
We probably need to handle the case of not finding crit_id here as well

Reply:
done added test associated with it